fennel/client/client.py

```diff
@@ -561,25 +550,14 @@ def extract_historical_features(
                     [f.fqn() for f in output_feature.features]
                 )
 
-        output_info = {}
-        if output_s3 is not None:
-            output_info["output_bucket"] = output_s3.bucket_name
-            output_info["output_prefix"] = output_s3.path_prefix
-            access_key_id, secret_access_key = output_s3.creds()
-            if access_key_id is not None:
-                output_info["output_access_key_id"] = access_key_id
-                if secret_access_key is None:
-                    raise Exception(
-                        "Output access key id specified but secret key not found."
-                    )
-                output_info["output_secret_access_key"] = secret_access_key
+        output_s3_table = _s3_connector_json(output_s3) if output_s3 else None
```

Comment:
I like having the helper function because it avoids duplicating the same logic for both the input and output buckets (about 25 lines) + only one place to change if we end up making any future changes to it

Reply:
I mean inline the var :)

Reply:
[sorry for bringing in too many nits, but just trying to bring in the best/recommended practices of languages I have picked up over few years at Google and being readability reviewer] :)